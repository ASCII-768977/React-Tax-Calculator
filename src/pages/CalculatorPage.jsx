import React from "react";
import Calculator from "../components/Calculator/Calculator";

const CalculatorPage = () => {
  return (
    <div id="background" data-testid="calculator-page">
      <Calculator />
    </div>
  );
};

export default CalculatorPage;
