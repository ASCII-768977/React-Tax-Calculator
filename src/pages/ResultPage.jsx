import React from "react";
import Result from "../components/Result/Result";

const ResultPage = () => {
  return (
    <div id="background" data-testid="result-page">
      <Result />
    </div>
  );
};

export default ResultPage;
