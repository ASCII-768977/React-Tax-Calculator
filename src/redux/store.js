import selectionReducer from "./reducer";
import { configureStore } from "@reduxjs/toolkit";

const store = configureStore({
  reducer: {
    selections: selectionReducer,
  },
});

export default store;
