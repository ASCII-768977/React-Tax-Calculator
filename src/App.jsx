import React from "react";
import { Route, Routes } from "react-router-dom";
import CalculatorPage from "./pages/CalculatorPage";
import ResultPage from "./pages/ResultPage";
import { Provider } from "react-redux";
import store from "./redux/store";

function App() {
  return (
    <Provider store={store}>
      <Routes>
        <Route path="/" element={<CalculatorPage />} />
        <Route path="/result" element={<ResultPage />} />
      </Routes>
    </Provider>
  );
}

export default App;
