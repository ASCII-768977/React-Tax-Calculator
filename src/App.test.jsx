import { render, screen } from "@testing-library/react";
import App from "./App";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import CalculatorPage from "./pages/CalculatorPage";
import ResultPage from "./pages/ResultPage";
import { Provider } from "react-redux";
import store from "./redux/store";

describe("App", () => {
  it("renders the calculator page by default", () => {
    render(<App />, { wrapper: MemoryRouter });
    const user = userEvent.setup();
    console.log("This is tree:");
    console.log(tree);
    console.log("This is tree type:");
    console.log(typeof tree);
    console.log("This is screen:");
    console.log(screen);
    console.log("This is screen type:");
    console.log(typeof screen);
    expect(screen.getByTestId("calculator-page")).toBeInTheDocument();
  });

  it("renders the result page when navigating to /result", () => {
    render(
      <MemoryRouter initialEntries={["/result"]}>
        <Provider store={store}>
          <Routes>
            <Route path="/" element={<CalculatorPage />} />
            <Route path="/result" element={<ResultPage />} />
          </Routes>
        </Provider>
      </MemoryRouter>
    );
    expect(screen.getByTestId("result-page")).toBeInTheDocument();
  });
});
