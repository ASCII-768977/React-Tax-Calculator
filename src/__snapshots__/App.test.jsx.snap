// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders the calculator page by default 1`] = `
<div
  data-testid="calculator-page"
  id="background"
>
  <section
    class="app-background"
  >
    <div
      class="left-section"
    >
      <h2
        class="left-section-heading"
      >
        Tax-o-tron
      </h2>
      <p
        class="left-section-description"
      >
        The free and simple online tax calculator
      </p>
      <div
        class="earth"
      />
      <div
        class="moon"
      />
    </div>
    <form
      action="#"
      class="right-form"
    >
      <h2
        class="form-heading"
      >
        Calculate your tax
      </h2>
      <div
        class="information"
      >
        <img
          alt="info"
          class="information-icon"
          src="/src/assets/information.svg"
        />
        <span
          class="information-text"
        >
          Fields marked with * are mandatory
        </span>
      </div>
      <label
        class="label-description"
        for="country"
      >
        Select your country of residence *
      </label>
      <select
        class="select-input"
      >
        <option
          value="Australia"
        >
          Australia
        </option>
        <option
          value="United States"
        >
          United States
        </option>
        <option
          value="China"
        >
          China
        </option>
        <option
          value="Canada"
        >
          Canada
        </option>
        <option
          value="New Zealand"
        >
          New Zealand
        </option>
      </select>
      <label
        class="label-description"
        for="year"
      >
        Select an income year *
      </label>
      <select
        class="select-input"
      >
        <option
          value="2020 - 2021"
        >
          2020 - 2021
        </option>
        <option
          value="2019 - 2020"
        >
          2019 - 2020
        </option>
        <option
          value="2018 - 2019"
        >
          2018 - 2019
        </option>
        <option
          value="2017 - 2018"
        >
          2017 - 2018
        </option>
      </select>
      <label
        class="label-description"
        for="amount"
      >
        Enter your total taxable income for the income year *
      </label>
      <div
        class="amount-input"
      >
        <div
          class="amount-input-sign"
        >
          $
        </div>
        <input
          class="amount-input-field"
          id="amount"
          name="amount"
          placeholder="Amount"
          type="number"
        />
        <div
          class="amount-input-foot"
        >
          .00
        </div>
      </div>
      <a
        href="/result"
      >
        <input
          class="submit"
          type="submit"
          value="Calculate"
        />
      </a>
    </form>
  </section>
</div>
`;
